### PR TITLE
🗳️ list elections with ballots

### LINKED ISSUE #176: ✨  Upload Ballot API
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

This is to ensure a ballot can be uploaded into the api for a specific open election. This should reject invalid ballots. Although this is just for the upload or post of the ballot. This card should also include the get and list so this can be tested. 

- [x] Ensure Ballot Upload API works with newer python 1.3 update. 

### Possible Implementation

_No response_

### Anything else?

_No response_

### LINKED ISSUE #202: 🐞 Sort elections by date descending
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The most recent is at the end, it should be at the beginning, when you findElection aka POST to `{{mediator-url}}/api/{{version}}/election/find?skip=0&limit=100` 

### Expected Behavior

The most recent should be at the top

### Steps To Reproduce

_No response_

### Environment

```markdown
- OS:
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lprichar**
### Issue
Fixes #176 
Fixes #202 

### Description
Adds a new endpoint for retrieving elections along with their name and the number of ballots they contain (both spoiled and cast).  Results sorted reverse chronologically.

### Testing

1. Authenticate
2. Hit {{mediator-url}}/api/{{version}}/election/list

Expected: a list of elections like:

```
{
    "status": "success",
    "message": null,
    "elections": [
        {
            "election_id": "hamilton-general-election-simple",
            "name": "Hamiltion County General Election",
            "state": "OPEN",
            "number_of_guardians": 5,
            "quorum": 3,
            "cast_ballots": 7,
            "spoiled_ballots": 2
        }
    ]
}
```

### CODE REVIEW

File: app/api/v1/mediator/election.py [hunk 1]
```diff
@@ -114,7 +128,57 @@ def create_election(
...
+    request: Request,
+) -> ElectionListResponseDto:
+    """
+    List all elections including state and number of ballots cast or submitted if any.
+    """
+
+    result = ElectionListResponseDto()
+    elections = filter_elections(filter={}, settings=request.app.state.settings)
+    result.elections = [to_election_summary(e) for e in elections]
+    index = 0
```
Comment (lprichar) [inline app/api/v1/mediator/election.py:161]: This indexing code is kinda ugly, but there are no existing system generated date fields that contain the creation date that I could find other than what's in the context, which is user provided and unreliable.  Would welcome ways to improve this.

File: app/api/v1/mediator/election.py [hunk 2]
```diff
@@ -114,7 +128,57 @@ def create_election(
...
+    for election in result.elections:
+        election.index = index
+        inventory = get_ballot_inventory(
```
Comment (lprichar) [inline app/api/v1/mediator/election.py:164]: This is an N+1 Selects problem, but there should be very few elections in the system, so I'm inclined to be lazy here unless anyone objects

File: app/api/v1/models/election.py [hunk 3]
```diff
@@ -100,6 +106,23 @@ class ElectionQueryResponse(BaseResponse):
     elections: List[Election] = []
 
 
+class ElectionSummaryDto(Base):
+    election_id: str
+    name: Any
+    state: str
+    number_of_guardians: int
+    quorum: int
+    cast_ballots: int
```
Comment (SteveMaier-IRT) [inline app/api/v1/models/election.py:115]: The ballot inventory uses cast_ballot_count and spoiled_ballot_count.  Should these be changed to be consistent?

**Review by SteveMaier-IRT — DISMISSED** [review]
Had a question in the comments, but overall looks good.

File: app/api/v1/models/election.py (hunk 3 above)
Comment (lprichar) [reply app/api/v1/models/election.py]: yup, good catch

**Review by SteveMaier-IRT — APPROVED** [review]
Looks good.

**Pull request merged**